### PR TITLE
feat(message): add helper methods to `Embed`

### PR DESCRIPTION
**Pull request by Bastien1533**
## About

This pull request is about the Embed Class that is used to create an embed. I have implemented methods for this class to manually change the embed after defining it.
The methods are the following:

```
    def add_field(self, name : str = None, value : str = None, inline : bool = True): ...
    def clear_fields(self): ...
    def insert_field_at(self, index : int,  name : str = None, value : str = None, inline : bool = True): ...
    def set_field_at(self, index : int = None, name : str = None, value : str = None, inline : bool = True): ...
    def remove_field(self, index): ...
    def remove_author(self): ...
    def set_author(self, name : str, url : str = None, icon_url : str = None, proxy_icon_url : str = None): ...
    def set_footer(self, text : str, icon_url : str = None, proxy_icon_url : str = None): ...
    def set_image(self, url : str, proxy_url : str = None, height : int = None, width : str = None): ...
    def set_thumbnail(self, url : str, proxy_url : str = None, height : int = None, width : str = None): ...
```

## Checklist

- [X] I've ran `pre-commit` to format and lint the change(s) made.
- [X] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent):
- [ ] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [X] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

**Review by sillyElvy — CHANGES_REQUESTED** [review]
Please add a docstring to every method

File: interactions/api/models/message.py [hunk 1]
```diff
@@ -1094,3 +1094,84 @@ def __init__(self, **kwargs):
 
         if self.footer:
             self._json.update({"footer": self.footer._json})
+
+    def add_field(self, name: str = None, value: str = None, inline: bool = True):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    def add_field(self, name: str, value: str, inline: Optional[bool] = False):
```
Name and value are required arguments, they should not have a default value. 

Inline is not present by default and not required, so please keep it to `False`


Also please keep in mind to use the `Optional` typehint for not required values

File: interactions/api/models/message.py [hunk 2]
```diff
@@ -1094,3 +1094,84 @@ def __init__(self, **kwargs):
...
+        if self.fields is None:
+            self.fields = []
+
+        self.fields.append(EmbedField(name=name, value=value, inline=inline))
+        self._json.update({"fields": [field._json for field in self.fields]})
+
+    def clear_fields(self):
+
+        self.fields = []
+        self._json.update({"fields": [field._json for field in self.fields]})
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
        self._json.update({"fields": []})
```

File: interactions/api/models/message.py [hunk 3]
```diff
@@ -1094,3 +1094,84 @@ def __init__(self, **kwargs):
...
+
+    def insert_field_at(self, index: int, name: str = None, value: str = None, inline: bool = True):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    def insert_field_at(self, index: int, name: str, value: str, inline: Optional[bool] = False):
```

File: interactions/api/models/message.py [hunk 4]
```diff
@@ -1094,3 +1094,84 @@ def __init__(self, **kwargs):
...
+        try:
+            self.fields.insert(index, EmbedField(name=name, value=value, inline=inline))
+
+        except AttributeError:
+            self.fields = [EmbedField(name=name, value=value, inline=inline)]
+
+        self._json.update({"fields": [field._json for field in self.fields]})
+
+    def set_field_at(
+        self, index: int = None, name: str = None, value: str = None, inline: bool = True
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
        self, index: int, name: str, value: str, inline: Optional[bool] = False
```

File: interactions/api/models/message.py [hunk 5]
```diff
@@ -1094,3 +1094,84 @@ def __init__(self, **kwargs):
...
+            self.fields[index] = EmbedField(name=name, value=value, inline=inline)
+            self._json.update({"fields": [field._json for field in self.fields]})
+
+        except AttributeError:
+            raise AttributeError("No fields found in Embed")
+
+        except IndexError:
+            raise IndexError("No fields at this index")
+
+    def remove_field(self, index):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    def remove_field(self, index: int):
```

File: interactions/api/models/message.py [hunk 6]
```diff
@@ -1094,3 +1094,84 @@ def __init__(self, **kwargs):
...
+    def remove_author(self):
+
+        try:
+            del self.author
+            self._json.update({"author": None})
+        except AttributeError:
+            pass
+
+    def set_author(
+        self, name: str, url: str = None, icon_url: str = None, proxy_icon_url: str = None
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
        self, name: str, url: Optional[str] = None, icon_url: Optional[str] = None, proxy_icon_url: Optional[str] = None
```

File: interactions/api/models/message.py [hunk 7]
```diff
@@ -1094,3 +1094,84 @@ def __init__(self, **kwargs):
...
+    ):
+
+        self.author = EmbedAuthor(
+            name=name, url=url, icon_url=icon_url, proxy_icon_url=proxy_icon_url
+        )
+        self._json.update({"author": self.author._json})
+
+    def set_footer(self, text: str, icon_url: str = None, proxy_icon_url: str = None):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    def set_footer(self, text: str, icon_url: Optional[str] = None, proxy_icon_url: Optional[str] = None):
```

File: interactions/api/models/message.py [hunk 8]
```diff
@@ -1094,3 +1094,84 @@ def __init__(self, **kwargs):
...
+
+        self.footer = EmbedFooter(text=text, icon_url=icon_url, proxy_icon_url=proxy_icon_url)
+        self._json.update({"footer": self.footer._json})
+
+    def set_image(self, url: str, proxy_url: str = None, height: int = None, width: str = None):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    def set_image(self, url: str, proxy_url: Optional[str] = None, height: Optional[int] = None, width: Optional[int] = None):
```

Why was width a string?

File: interactions/api/models/message.py [hunk 9]
```diff
@@ -1094,3 +1094,84 @@ def __init__(self, **kwargs):
...
+
+        self.image = EmbedImageStruct(url=url, proxy_url=proxy_url, height=height, width=width)
+        self._json.update({"image": self.image._json})
+
+    def set_thumbnail(self, url: str, proxy_url: str = None, height: int = None, width: str = None):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    def set_thumbnail(self, url: str, proxy_url: Optional[str] = None, height: Optional[int] = None, width: Optional[int] = None):
```

File: interactions/api/models/message.pyi [hunk 10]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
     author: Optional[EmbedAuthor]
     fields: Optional[List[EmbedField]]
     def __init__(self, **kwargs): ...
+    def add_field(self, name : str = None, value : str = None, inline : bool = True): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def add_field(self, name: str, value: str, inline: Optional[bool] = False): ...
```
No whitespace before a colon

File: interactions/api/models/message.pyi [hunk 11]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def clear_fields(self): ...
+    def insert_field_at(self, index : int,  name : str = None, value : str = None, inline : bool = True): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def insert_field_at(self, index: int,  name: str, value: str, inline: Optional[bool] = False): ...
```

File: interactions/api/models/message.pyi [hunk 12]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def set_field_at(self, index : int = None, name : str = None, value : str = None, inline : bool = True): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def set_field_at(self, index: int, name: str, value: str, inline: Optional[bool] = False): ...
```

File: interactions/api/models/message.pyi [hunk 13]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def remove_field(self, index): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def remove_field(self, index: int): ...
```

File: interactions/api/models/message.pyi [hunk 14]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def remove_author(self): ...
+    def set_author(self, name : str, url : str = None, icon_url : str = None, proxy_icon_url : str = None): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def set_author(self, name: str, url: Optional[str] = None, icon_url: Optional[str] = None, proxy_icon_url: Optional[str] = None): ...
```

File: interactions/api/models/message.pyi [hunk 15]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def set_footer(self, text : str, icon_url : str = None, proxy_icon_url : str = None): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def set_footer(self, text: str, icon_url: Optional[str] = None, proxy_icon_url: Optional[str] = None): ...
```

File: interactions/api/models/message.pyi [hunk 16]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def set_image(self, url : str, proxy_url : str = None, height : int = None, width : str = None): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def set_image(self, url: str, proxy_url: Optional[str] = None, height: Optional[int] = None, width: Optional[int] = None): ...
```

File: interactions/api/models/message.pyi [hunk 17]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def set_thumbnail(self, url : str, proxy_url : str = None, height : int = None, width : str = None): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def set_thumbnail(self, url: str, proxy_url: Optional[str] = None, height: Optional[int] = None, width: Optional[int] = None): ...
```

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/api/models/message.py (hunk 1 above)
Comment (Bastien1533) [reply interactions/api/models/message.py]: Oh yeah my bad for the name and value, however for the inline, I think it should be by default to True ?

Comment (sillyElvy) [reply interactions/api/models/message.py]: It is not present by default, so it is either `None` or `False`

Comment (Bastien1533) [reply interactions/api/models/message.py]: Ok, I was used to dpy defaulting to True.

Comment (sillyElvy) [reply interactions/api/models/message.py]: Set it to none then, so we don't send any value for it

Comment (Bastien1533) [reply interactions/api/models/message.py]: Ok

**Comment by i0bs** [comment]
I'm not sure what makes it beneficial to have these helper methods over directly modifying the attributes in the ``Embed`` object itself. A majority, if not all of these, can simply be done with direct writing.

I would need to see some more use cases to find a reason worthy for adding this, but I will labeling this as a wontfix until then.

**Comment by Bastien1533** [comment]
Hey, I think these helpers are an easy and comprehensive way of dynamically changing embeds. What has the most utility I think are the fields helpers. It provides a clean way of appending new Fields and directly updating the embed's json, for example:  `add_field(...)`. The other methods, may not be as useful as the fields ones but follow the same "dynamic" way of doing things.

**Comment by i0bs** [comment]
So what I'm gathering is the intuitiveness to use helper methods should be strongly considered. In fairness, embeds could be """dynamically""" updated in their fields by appending the list. Would it be better to use `add_field(...)` instead of `embed.fields.append(EmbedField(...))`?

**Comment by Bastien1533** [comment]
`embed.add_field()` is way "cleaner" as it underlies the `EmbedField()` Model and also the most important thing : it updates the `._json`. Just appending the `EmbedField()` to `fields `doesn't update the `._json`. I have also talked to many people that would be interested in those methods.

**Comment by mAxYoLo01** [comment]
From personal experience, I can clearly say that `add_field` and other such methods are definitely helpful to add in this library, just to simplify the library's usage to future developers.
In fact, I think that a lot of other classes would benefit from adding these types of helper methods. Plus, it really takes almost no time to develop these features, and can easily be done by other contributors!

**Comment by FayeDel** [comment]
In terms of potential usecases, I can see this being more dynamic than weird ternary logic with embed fields on instantiation.
A usecase I can think of would be adding fields depending on command-given attributes (like referencing an Embed across a command instance, and dynamically add fields by command arguments given).

As per Bastien's point, it's easier than adding it to a direct list because of updating the _json attribute (Because it doesn't autoupdate if we modify the list object directly, implementing so would tend to give a lot more complications than necessary)

File: interactions/api/models/message.py [hunk 18]
```diff
@@ -901,3 +901,194 @@ def __init__(self, **kwargs):
 
         if self.footer:
             self._json.update({"footer": self.footer._json})
+
+    def add_field(self, name: str, value: str, inline: Optional[bool] = False):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    def add_field(self, name: str, value: str, inline: Optional[bool] = False) -> None:
```

File: interactions/api/models/message.py [hunk 19]
```diff
@@ -901,3 +901,194 @@ def __init__(self, **kwargs):
...
+        :type inline?: Optional[bool]
+        """
+
+        if self.fields is None:
+            self.fields = []
+
+        self.fields.append(EmbedField(name=name, value=value, inline=inline))
+        self._json.update({"fields": [field._json for field in self.fields]})
+
+    def clear_fields(self):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    def clear_fields(self) -> None:
```

File: interactions/api/models/message.py [hunk 20]
```diff
@@ -901,3 +901,194 @@ def __init__(self, **kwargs):
...
+        """
+        Clears all the fields of the embed
+        """
+
+        self.fields = []
+        self._json.update({"fields": []})
+
+    def insert_field_at(
+        self, index: int, name: str = None, value: str = None, inline: Optional[bool] = False
+    ):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    ) -> None:
```

File: interactions/api/models/message.py [hunk 21]
```diff
@@ -901,3 +901,194 @@ def __init__(self, **kwargs):
...
+
+        try:
+            self.fields.insert(index, EmbedField(name=name, value=value, inline=inline))
+
+        except AttributeError:
+            self.fields = [EmbedField(name=name, value=value, inline=inline)]
+
+        self._json.update({"fields": [field._json for field in self.fields]})
+
+    def set_field_at(self, index: int, name: str, value: str, inline: Optional[bool] = False):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    def set_field_at(self, index: int, name: str, value: str, inline: Optional[bool] = False) -> None:
```

File: interactions/api/models/message.py [hunk 22]
```diff
@@ -901,3 +901,194 @@ def __init__(self, **kwargs):
...
+            self.fields[index] = EmbedField(name=name, value=value, inline=inline)
+            self._json.update({"fields": [field._json for field in self.fields]})
+
+        except AttributeError:
+            raise AttributeError("No fields found in Embed")
+
+        except IndexError:
+            raise IndexError("No fields at this index")
+
+    def remove_field(self, index):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    def remove_field(self, index: int) -> None:
```

File: interactions/api/models/message.py [hunk 23]
```diff
@@ -901,3 +901,194 @@ def __init__(self, **kwargs):
...
+            self.fields.pop(index)
+            self._json.update({"fields": [field._json for field in self.fields]})
+
+        except AttributeError:
+            raise AttributeError("No fields found in Embed")
+
+        except IndexError:
+            raise IndexError("Field not Found at index")
+
+    def remove_author(self):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    def remove_author(self) -> None:
```

File: interactions/api/models/message.py [hunk 24]
```diff
@@ -901,3 +901,194 @@ def __init__(self, **kwargs):
...
+        except AttributeError:
+            pass
+
+    def set_author(
+        self,
+        name: str,
+        url: Optional[str] = None,
+        icon_url: Optional[str] = None,
+        proxy_icon_url: Optional[str] = None,
+    ):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    ) -> None:
```

File: interactions/api/models/message.py [hunk 25]
```diff
@@ -901,3 +901,194 @@ def __init__(self, **kwargs):
...
+        """
+
+        self.author = EmbedAuthor(
+            name=name, url=url, icon_url=icon_url, proxy_icon_url=proxy_icon_url
+        )
+        self._json.update({"author": self.author._json})
+
+    def set_footer(
+        self, text: str, icon_url: Optional[str] = None, proxy_icon_url: Optional[str] = None
+    ):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    ) -> None:
```

File: interactions/api/models/message.py [hunk 26]
```diff
@@ -901,3 +901,194 @@ def __init__(self, **kwargs):
...
+        self.footer = EmbedFooter(text=text, icon_url=icon_url, proxy_icon_url=proxy_icon_url)
+        self._json.update({"footer": self.footer._json})
+
+    def set_image(
+        self,
+        url: str,
+        proxy_url: Optional[str] = None,
+        height: Optional[int] = None,
+        width: Optional[int] = None,
+    ):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    ) -> None:
```

File: interactions/api/models/message.py [hunk 27]
```diff
@@ -901,3 +901,194 @@ def __init__(self, **kwargs):
...
+        self.image = EmbedImageStruct(url=url, proxy_url=proxy_url, height=height, width=width)
+        self._json.update({"image": self.image._json})
+
+    def set_thumbnail(
+        self,
+        url: str,
+        proxy_url: Optional[str] = None,
+        height: int = None,
+        width: Optional[str] = None,
+    ):
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    ) -> None:
```

File: interactions/api/models/message.pyi [hunk 28]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
     author: Optional[EmbedAuthor]
     fields: Optional[List[EmbedField]]
     def __init__(self, **kwargs): ...
+    def add_field(self, name: str, value: str, inline: Optional[bool] = False): ...
+    def clear_fields(self): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def clear_fields(self) -> None: ...
```

File: interactions/api/models/message.pyi [hunk 29]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
     author: Optional[EmbedAuthor]
     fields: Optional[List[EmbedField]]
     def __init__(self, **kwargs): ...
+    def add_field(self, name: str, value: str, inline: Optional[bool] = False): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def add_field(self, name: str, value: str, inline: Optional[bool] = False) -> None: ...
```

File: interactions/api/models/message.pyi [hunk 30]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def clear_fields(self): ...
+    def insert_field_at(self, index: int,  name: str, value: str, inline: Optional[bool] = False): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def insert_field_at(self, index: int,  name: str, value: str, inline: Optional[bool] = False) -> None: ...
```

File: interactions/api/models/message.pyi [hunk 31]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def set_field_at(self, index: int, name: str, value: str, inline: Optional[bool] = False): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def set_field_at(self, index: int, name: str, value: str, inline: Optional[bool] = False) -> None: ...
```

File: interactions/api/models/message.pyi [hunk 32]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def remove_field(self, index: int): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def remove_field(self, index: int) -> None: ...
```

File: interactions/api/models/message.pyi [hunk 33]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def remove_author(self): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def remove_author(self) -> None: ...
```

File: interactions/api/models/message.pyi [hunk 34]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def set_author(self, name: str, url: Optional[str] = None, icon_url: Optional[str] = None, proxy_icon_url: Optional[str] = None): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def set_author(self, name: str, url: Optional[str] = None, icon_url: Optional[str] = None, proxy_icon_url: Optional[str] = None) -> None: ...
```

File: interactions/api/models/message.pyi [hunk 35]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def set_footer(self, text: str, icon_url: Optional[str] = None, proxy_icon_url: Optional[str] = None): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def set_footer(self, text: str, icon_url: Optional[str] = None, proxy_icon_url: Optional[str] = None) -> None: ...
```

File: interactions/api/models/message.pyi [hunk 36]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def set_image(self, url: str, proxy_url: Optional[str] = None, height: Optional[int] = None, width: Optional[int] = None): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def set_image(self, url: str, proxy_url: Optional[str] = None, height: Optional[int] = None, width: Optional[int] = None) -> None: ...
```

File: interactions/api/models/message.pyi [hunk 37]
```diff
@@ -244,3 +244,13 @@ class Embed(DictSerializerMixin):
...
+    def set_thumbnail(self, url: str, proxy_url: Optional[str] = None, height: Optional[int] = None, width: Optional[int] = None): ...
```
Comment (sillyElvy) [inline interactions/api/models/message.pyi]:
```suggestion
    def set_thumbnail(self, url: str, proxy_url: Optional[str] = None, height: Optional[int] = None, width: Optional[int] = None) -> None: ...
```

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/api/models/message.py (hunk 18 above)
Comment (sillyElvy) [reply interactions/api/models/message.py]: You did not change the code

File: interactions/api/models/message.py [hunk 38]
```diff
@@ -902,11 +902,195 @@ def __init__(self, **kwargs):
...
+            self.fields[index] = EmbedField(name=name, value=value, inline=inline)
+            self._json.update({"fields": [field._json for field in self.fields]})
+
+        except AttributeError:
+            raise AttributeError("No fields found in Embed")
+
+        except IndexError:
+            raise IndexError("No fields at this index")
+
+    def remove_field(self, index) -> None:
```
Comment (sillyElvy) [inline interactions/api/models/message.py]:
```suggestion
    def remove_field(self, index: int) -> None:
```

File: interactions/api/models/message.py [hunk 39]
```diff
@@ -902,11 +902,195 @@ def __init__(self, **kwargs):
...
+        """
+        Remove field at the specified index
+
+        :param index: The new field's index
+        :type index: int
+        """
+
+        try:
+            self.fields.pop(index)
+            self._json.update({"fields": [field._json for field in self.fields]})
```
Comment (sillyElvy) [inline interactions/api/models/message.py:991]: Check if fields is empty please

**Review by sillyElvy — CHANGES_REQUESTED** [review]

**Review by sillyElvy — COMMENTED** [review]
I think this LGTM now :)

**Review by sillyElvy — APPROVED** [review]
(ok bruh i misclicked now its approved lmao)

**Pull request merged**